File: Java/opentraceability/src/main/java/opentraceability/gdst/events/GDSTHatchingEvent.java
package opentraceability.gdst.events;

import opentraceability.Constants;
import opentraceability.models.events.EventAction;
import opentraceability.models.events.ObjectEventBase;
import opentraceability.models.identifiers.PGLN;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

import java.net.URI;

public class GDSTHatchingEvent extends ObjectEventBase<GDSTILMD> implements IGDSTEvent {
    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "productOwner")
    @OpenTraceabilityJsonAttribute(name = "gdst:productOwner")
    private PGLN productOwner;

    @Override
    public PGLN getProductOwner() {
        return this.productOwner;
    }


    @OpenTraceabilityAttribute(ns = Constants.GDST_NAMESPACE, name = "humanWelfarePolicy")
    @OpenTraceabilityJsonAttribute(name = "gdst:humanWelfarePolicy")
    private String humanWelfarePolicy;

    // Add constructor, getters, and setters here
    public GDSTHatchingEvent()
    {
        this.businessStep = URI.create("urn:epcglobal:cbv:bizstep:hatching");
        this.action = EventAction.ADD;
    }
}

File: CSharp/OpenTraceability.MSC/Events/Gdst12/GDSTHatchingEvent.cs
using OpenTraceability.GDST.Events;
using OpenTraceability.Models.Events;
using OpenTraceability.Models.Identifiers;
using OpenTraceability.Utility.Attributes;

namespace OpenTraceability.MSC.Events
{
    /// <summary>
    /// A hatching event as defined by GDST 1.2: OBJECT / ADD / business step "urn:gdst:bizStep:hatching".
    /// GDST 2.0 expresses it as a commission event on a product classified as developing.
    /// </summary>
    public class GDSTHatchingEvent : ObjectEvent<GDSTILMD>, IGDSTILMDEvent
    {
        [OpenTraceability(Constants.GDST_NAMESPACE, "productOwner")]
        [OpenTraceabilityJson("gdst:productOwner")]
        public PGLN? ProductOwner { get; set; }

        [OpenTraceability(Constants.GDST_NAMESPACE, "humanWelfarePolicy")]
        [OpenTraceabilityJson("gdst:humanWelfarePolicy")]
        public string? HumanWelfarePolicy { get; set; }

        public GDSTHatchingEvent()
        {
            this.BusinessStep = new Uri("urn:gdst:bizStep:hatching");
            this.Action = EventAction.ADD;
        }
    }
}
